rdtools/normalization.py

```diff
@@ -383,40 +375,270 @@ def check_series_frequency(series, series_description):
+    '''return a series of right labeled differences in the index of time_series in nanoseconds'''
+    t_steps = np.diff(time_series.index.astype('int64')).astype('float')
+    t_steps = np.insert(t_steps, 0, np.nan)
+    t_steps = pd.Series(index=time_series.index, data=t_steps)
+    return t_steps
+
+
+def energy_from_power(time_series, target_frequency=None, max_timedelta=None):
     '''
-    Returns a right-labeled energy time series from an instantaneous power time series.
-    Energy is not returned when the gap between data points exceeds max_interval_hours.
-    power_series should be given in Watts, and max_interval_hours in hours.
+    Returns a regular right-labeled energy time series in units of Wh per interval from an
+    instantaneous power time series. NaN is filled where the gap between input data points
+    exceeds max_timedelta. Power_series should be given in Watts.
 
     Parameters
     ----------
-    power_series: Pandas Series with DatetimeIndex
-        Instantaneous time series of PV power generation in Watts
-    max_interval_hours: float or NoneType (default: None)
+    time_series: Pandas Series with DatetimeIndex
+        Instantaneous time series of power in Watts
+    target_frequency: DatetimeOffset, or frequency string
+        The frequency of the energy time series to be returned. If None(default),
+        the fequency is set to the median time step in time_series
+    max_timedelta: Timedelta or NoneType (default: None)
         The maximum allowed gap between power measurements. If the gap between
-        consecutive power measurements exceeds max_interval_hours, no energy value
-        will be returned for that interval. If None, max_interval_hours is set internally
-        to the median time delta in power_series and a UserWarning is issued.
+        consecutive power measurements exceeds max_timedelta, NaN
+        will be returned for that interval. If None, max_timedelta is set internally
+        to the median time delta in time_series.
 
     Returns:
     --------
-    right-labeled energy pandas time series in Wh
-
+    right-labeled energy pandas time series in Wh per interval
     '''
 
-    if not isinstance(power_series.index, pd.DatetimeIndex):
+    if not isinstance(time_series.index, pd.DatetimeIndex):
         raise ValueError('power_series must be a pandas series with a DatetimeIndex')
 
-    times = np.array(power_series.index.astype('int64'))
-    time_deltas = np.diff(times) / 10.0**9 / 3600.0
+    t_steps = t_step_nanoseconds(time_series)
+    median_step_ns = t_steps.median()
+
+    if target_frequency is None:
+        target_frequency = str(median_step_ns) + 'N'  # Pandas offset allias in ns
+
+    if max_timedelta is None:
+        max_interval_nanoseconds = median_step_ns
+    else:
+        max_interval_nanoseconds = max_timedelta.total_seconds() * 10.0**9
+
+    try:
+        freq_interval_size_ns = pd.tseries.frequencies.to_offset(target_frequency).nanos
```

Comment:
I'm getting an error here when I run the example notebook: 
`df['energy'] = rdtools.energy_from_power(df.power, max_timedelta=pd.to_timedelta('15 minutes'))`

Error message:

>ValueError: Invalid frequency: 3e+11N